### PR TITLE
add/a2p api-core-tests product properties tests

### PR DESCRIPTION
**Pull request by nigeljamesstevenson**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adding CRUD tests for the tax rates and tax classes to ensure we have better test coverage
Closes 366-gh-woocommerce/woocommerce-quality and 363-gh-woocommerce/woocommerce-quality.

### How to test the changes in this Pull Request:

1. Pull down the code
2. Navigate to woocommerce/plugins/woocommerce
3. Execute tests with USE_WP_ENV=1 pnpm playwright test --config=tests/api-core-tests/playwright.config.js tests/api-core-tests/tests/products/products-crud.test.js
4. The products tests should execute and pass successfully including the new tags, attributes and attributes terms tests

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js [hunk 1]
```diff
@@ -18,118 +23,1084 @@ const { batch } = require( '../../data/shared/batch-update' );
...
+				`wp-json/wc/v3/products/tags/batch`, {
+					data: {
+						delete: [attributeId1, attributeId3]
+					}
+				}
+			);
+		});
+	});
+
+	// test.describe('Product **variations** tests: CRUD', () => {
```
Comment (zhongruige) [inline plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js]: Is there a reason we have these tests commented out here? Could we put these in a `skip` block instead, or should we just remove them?

**Review by zhongruige — CHANGES_REQUESTED** [review]
Thank you for working on this @nigeljamesstevenson!

These ran great for me locally:

Screen Shot 2022-10-07 at 11 57 59

Left a comment on the commented out section of tests, if they should be removed, or put in a `skip` block--or if we need a follow up issue for them. Once that's answered this should be all set to merge.

Thanks!

File: plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js (hunk 1 above)
Comment (nigeljamesstevenson) [reply plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js]: These commented out tests should now be removed - thanks for pointing this out!

**Review by zhongruige — APPROVED** [review]
Changes look great! Thank you for working on this @nigeljamesstevenson!

**Pull request merged**